### PR TITLE
replace is_active to determine available xmlrpc methods

### PR DESCRIPTION
**Pull request by leogermani**
Stop using is_active to determine what XMLRPC endpoints should be available for the requests.

This is part of the cleanup to slowly deprecate the `is_active` method and open the path to the user-less project.

#### Changes proposed in this Pull Request:

* Changes the criteria used to register XML-RPC methods

#### Jetpack product discussion

1191179647901802-as-1200087466875449

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

In order to test this PR, there is this handy script -> 28d59-pb/#php

It will make 3 request to `system.listMethods` in your site. Save this script in your sandbox `public_html` folder and edit it to add your test site ID.

* One with an unauthenticated request
* One authenticated with a blog token
* One authenticated with a user token

So the test flow is:

* Disconnect your site
* Run the script on your sandbox
* Connect only at a site level
* Run the script
* Connect  a user
* Run the script

This will give you 3 outputs. Each output with the three types of request.

Look at the table here -> p9dueE-1Wm-p2

Make sure the exposed methods match the table. Each of the script runs will represent one column in the table.

Also, please read the code, follow the logic and check the criteria are not altered.

Finally, make sure the Jetpack Debugger passes all the tests and test the connection flows.

**NOTE** For the Debugger tests to pass you'll need https://github.com/Automattic/jetpack/pull/19301 and D59309-code (both deployed/merged already)

### CODE REVIEW

File: projects/packages/connection/src/class-manager.php [hunk 1]
```diff
@@ -149,30 +151,28 @@ public function setup_xmlrpc_handlers(
...
-		if ( $is_active ) {
+		if ( $is_signed ) {
+			// If the site is connected either at a site or user level and the request is signed, expose the methods.
+			// The callback is responsible to determine whether the request is signed with blog or user token and act accordingly.
+			// The actual API methods.
+			$callback = array( $this->xmlrpc_server, 'xmlrpc_methods' );
+
 			// Hack to preserve $HTTP_RAW_POST_DATA.
-			add_filter( 'xmlrpc_methods', array( $this, 'xmlrpc_methods' ) );
+			add_filter( 'xmlrpc_methods', array( $this->xmlrpc_server, 'xmlrpc_methods' ) );
```
Comment (fgiannar) [inline projects/packages/connection/src/class-manager.php:161]:
```suggestion
			add_filter( 'xmlrpc_methods', array( $this, 'xmlrpc_methods' ) );
```

typo?

**Review by fgiannar — CHANGES_REQUESTED** [review]
Hey Leo!

Great work here! Thanks for providing a handy script to test with!

Code looks good, logic is solid I believe.
I've found a typo (see inline comments).

I'm having a bit of trouble testing with the provided script. I seem to get `NULL` for unauthorized requests to a non connected site.
I've tried it via Postman though and works as expected.

**Comment by leogermani** [comment]
@fgiannar Thanks for spotting the typo.

I've fixed it and also updating the broken tests. As soon as https://github.com/Automattic/jetpack/pull/18952 is merged, I'll move these tests into the Connection package, where they belong.

As for the testing script, I understand you managed to test it with Postman, right?

**Comment by leogermani** [comment]
Updated the testing script with one that should work for all sites: 28d59-pb/#php

(also updated in the PR description)

**Review by fgiannar — APPROVED** [review]
Looks great, Leo! I tested using the provided instructions and everything works as expected! 💯

**Pull request merged**